packages/components/pie-toggle-switch/src/defs.ts

```diff
@@ -1,3 +1,19 @@
-// TODO - please remove the eslint disable comment below when you add props to this interface
-// eslint-disable-next-line @typescript-eslint/no-empty-interface
-export interface ToggleSwitchProps {}
+import { RTLComponentProps } from '@justeattakeaway/pie-webc-core';
+
+export interface ToggleSwitchProps extends RTLComponentProps {
+    /**
+     * Same as the HTML checked attribute - indicates whether the switch is on or off
+     */
+    isChecked: boolean;
+    /**
+     * Same as the HTML checked attribute - indicates whether the switch disabled or not
+     */
+    isDisabled: boolean;
+}
```

Comment:
I've previously seen some linter warnings about "cannot use namespace X as a type". I think it was in the story files, do you see the same?

Either way, I think semantically `type` makes more sense for these prop types than `interface`.

```suggestion
export type ToggleSwitchProps = RTLComponentProps & {
    /**
     * Same as the HTML checked attribute - indicates whether the switch is on or off
     */
    isChecked: boolean;
    /**
     * Same as the HTML checked attribute - indicates whether the switch disabled or not
     */
    isDisabled: boolean;
};
```

Reply:
Out of interest, why is types better? I thought that interfaces provides a simpler way to reuse code. Types seem like it should only be used if the model is complex i.e union, intersection mapped etc. Or are we saying that this could potentially use those in future therefore use types from the start?